commit c72325d52575cb8a46cc8595c592c27581089dec
Author: Takao Fujiwara <tfujiwar@redhat.com>
Date:   Tue Mar 24 19:08:34 2015 +0900

    ibus: Get display number when screen number is omitted.
    
    QIBusPlatformInputContextPrivate::createConnection() parses DISPLAY
    variable and it does not get the display number when the screen number
    is omitted. E.g. DISPLAY=":1".
    
    Change-Id: I9f81d8114d4e0f654a27ada67c5dc110eb34cd64
    Reviewed-by: Gatis Paeglis <gatis.paeglis@digia.com>

diff --git a/src/plugins/platforminputcontexts/ibus/qibusplatforminputcontext.cpp b/src/plugins/platforminputcontexts/ibus/qibusplatforminputcontext.cpp
index d00435b..673942b 100644
--- a/src/plugins/platforminputcontexts/ibus/qibusplatforminputcontext.cpp
+++ b/src/plugins/platforminputcontexts/ibus/qibusplatforminputcontext.cpp
@@ -351,6 +351,8 @@ QDBusConnection *QIBusPlatformInputContextPrivate::createConnection()
     int pos2 = display.indexOf('.', pos);
     if (pos2 > 0)
         displayNumber = display.mid(pos, pos2 - pos);
+    else
+        displayNumber = display.right(pos);
     if (debug)
         qDebug() << "host=" << host << "displayNumber" << displayNumber;
 
